subsys/input/input_longpress.c

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT zephyr_input_longpress
+
+#include <zephyr/device.h>
+#include <zephyr/input/input.h>
+#include <zephyr/kernel.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(input_longpress, CONFIG_INPUT_LOG_LEVEL);
+
+struct longpress_config {
+	const struct device *input_dev;
+	const uint16_t *input_codes;
+	const uint16_t *short_codes;
+	const uint16_t *long_codes;
+	const uint32_t long_delays_ms;
+	const uint8_t num_codes;
```

Comment:
nit: const not needed here

Reply:
ok dropping it, misunderstood the "fixed" in the previous comment.